core/modules/wsched/ChunkDisk.cc

```diff
@@ -101,13 +101,10 @@ void ChunkDisk::enqueue(wbase::Task::Ptr const& a) {
         _activeTasks.push(a);
         state = "EMPTY";
     } else {
-        if(chunkId < _chunkState.lastScan()) {
+        // To keep from getting stuck  on this chunkId, put new requests for this chunk on pending.
+        if(chunkId <= _chunkState.lastScan()) {
```

Comment:
This is the only change in behavior that really shows. The ChunkStatus cache was effectively nullified by putting anything with chunkId < _chunkState().lastScan() on pending as anything in the cache is going to be less than lastScan, except when wrapping around. This, and some other problems were not apparent until I started writing the tests and it's hard to split them apart.